### DISCUSSION TITLE
Substituting latex command in equation and math environment

### DISCUSSION THREAD

**Discussion by deb75** [jgm/pandoc#10486] [Q&A] [answered]
I use pandoc to convert from org file to docx.

It works quite well except that I have some defined latex commands that are not translated in the finale docx document.
Instead, they are treated as raw text.

I would like to use filters to change on the fly my latex commands to something pandoc understands.

As an example, `\ket{q}` is not understood, but `\vert q \ket` is correctly translated.

I have difficulties to understand how pandoc lua filters work. How would you do with this simple example ?

Regards

**Comment by jgm** [comment]
simple example:
```lua
function Math(el)
  el.text = string.gsub(el.text, "\\ket", "something else")
end
```

**Comment by deb75** [comment, marked as answer]
Thanks for your answer.

I wrote this
```lua
function Math(el)
    el.text = string.gsub(el.text, "\\ket{(.-)}", "\\vert %1 \\rangle")
end
```
in a file `filter-math.lua`.

Then, I tested it with : 
```bash
printf "$\ket{eaaza}$ hello" | pandoc --from latex -t native  --lua-filter filter-math.lua
```

It returned : 
```
[ Para
    [ Math InlineMath "\\ket{eaaza}" , Space , Str "hello" ]
]
```

Pandoc did not return any errors (exit code=0). What am i doting wrong ?

EDIT1 : I checked directly in lua interpreter that the lua code works as expected, it seems that `pandoc` just not applies it.

EDIT2: Got it ! You must return the element `el` at the end of lua function. So the working code is : 

``` lua
function Math(el)
    el.text = string.gsub(el.text, "\\ket{(.-)}", "\\vert %1 \\rangle")
    return el
end
```

If you have several replacements to do in math environments, do you have to make them in the same function `Math` ?
Regards

**Reply by jgm** [reply]
Sorry I forgot that part, glad you figured it out.

> If you have several replacements to do in math environments, do you have to make them in the same function Math ?

Yes.
